### ISSUE TITLE
Support new character sets in MySQL 8.0

### ISSUE THREAD

**Issue by bgrainger** [mysql-net/MySqlConnector#242]
Run the SideBySide tests against MySQL Server 8.0.1; `DataTypes.QueryBinaryGuid(false)` fails:

```
Failed   SideBySide.DataTypes.QueryBinaryGuid(oldGuids: False)
Error Message:
 Assert.Equal() Failure
Expected: typeof(System.Guid)
Actual:   typeof(string)
Stack Trace:
   at SideBySide.DataTypes.QueryBinaryGuid(Boolean oldGuids) in MySqlConnector\tests\SideBySide\DataTypes.cs:line 346
```

Note that this failure doesn't happen when running *just* the `QueryBinaryGuid` tests, nor when setting `csb.MaximumPoolSize = 1234;` in the test, which indicates it may be related to pulling an existing connection from the pool.

The failure also doesn't reproduce when using MySql.Data 6.9.9.

**Comment by bgrainger** [comment]
For a passing test, the values on [line 337](https://github.com/mysql-net/MySqlConnector/blob/8a54bdd48e032b15f1130862b88fa3f452abf8f1/src/MySqlConnector/MySqlClient/Results/ResultSet.cs#L337) are:

ColumnLength | CharacterSet | BytesPerCharacter | Computed Length | Correct
--- | --- | --- | --- | ---
144 | Utf8Mb4Binary | 4 | 36 | ✔️

For a failing test, they are:

ColumnLength | CharacterSet | BytesPerCharacter | Computed Length | Correct
--- | --- | --- | --- | ---
144 | 255 | 1 | 144 | ❌

It appears that MySQL 8.0 has new characters sets that we need to support. Specifically, `SELECT id, collation_name FROM information_schema.collations WHERE id  > 250 ORDER BY id;` lists 43 new character sets.

**Comment by bgrainger** [comment]
Fixed in [0.18.1](https://www.nuget.org/packages/MySqlConnector/0.18.1).
